### PR TITLE
Docusaurus2 Migration

### LINKED ISSUE #564: Migrate Website to Docusaurus 2 Alpha
### Description

Hi there! The Docusaurus team is currently developing Docusaurus 2 and are almost done with an MVP that doesn't include translations and versioning features, which means there's feature parity with your website's current usage of Docusaurus.

Docusaurus 2 brings about many improvements:

- Client-side rendering with prerendering (site renders without JavaScript!)
- More flexible appearance - Customizable layouts, CSS modules, etc
- Embeddable interactive React components within markdown via MDX

I will be your POC in helping you throughout the migration process. Let me know what you think!

cc @yangshun @endiliey @wgao19

### PR DESCRIPTION
**Pull request by scottilee**
## Motivation

Migrates idb website to Docusaurus v2. This [issue](https://github.com/facebook/docusaurus/issues/1834) has more information. Additionally, there are some doc fixes.

Fixes #564 

## Test Plan

Try it out here: https://idb.scottlee.now.sh

v2 screenshot:
Screen Shot 2020-02-22 at 11 13 06 AM

## Related PRs

N/A

/cc @yangshun

### CODE REVIEW

**Comment by scottilee** [comment]
@yangshun there is still an issue where the test-execution doc page sidebar is showing at the bottom of the page.

File: docs/architecture.md [hunk 1]
```diff
@@ -2,15 +2,17 @@
...
+This is a python3 cli that exposes all of the functionality that idb has to offer. As it is written in Python, we can run this basically anywhere.
 The cli itself just wraps a client class which can also be imported elsewhere. It talks to an idb daemon over gRPC.
 
 ## The idb_companion
 
 The companion is a gRPC server in Objective-C and C++. It talks to the native APIs that are used for automating Simulators and Devices. It links FBSimulatorControl and FBDeviceControl to perform these tasks.
 The companion is paired with a single iOS target (a device/simulator), handles executing the requested commands.
 
-![](assets/idb_architecture.png)
+![](https://www.fbidb.io/docs/assets/idb_architecture.png)
```
Comment (yangshun) [inline docs/architecture.md]: Better to use [`useBaseUrl()`](https://v2.docusaurus.io/docs/static-assets/#markdown-example) for better maintainability.

File: website/docusaurus.config.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// See https://docusaurus.io/docs/site-config for all the possible
```
Comment (yangshun) [inline website/docusaurus.config.js]: This site is now v2 so this comment is inaccurate. Let's remove the entire comment.

File: website/package.json [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
+    "build": "docusaurus build",
+    "swizzle": "docusaurus swizzle",
+    "deploy": "docusaurus deploy"
   },
   "license": "MIT",
-  "devDependencies": {
-    "docusaurus": "^1.8.1"
+  "dependencies": {
+    "@docusaurus/core": "^2.0.0-alpha.43",
+    "@docusaurus/preset-classic": "^2.0.0-alpha.40",
```
Comment (yangshun) [inline website/package.json]: We need to upgrade both at the same time. Always use the same version for both.

File: website/src/css/custom.css [hunk 4]
```diff
@@ -0,0 +1,9 @@
+:root {
```
Comment (yangshun) [inline website/src/css/custom.css]: Add the copyright header (see lint warning on D20065823).

File: docs/architecture.md (hunk 1 above)
Comment (scottilee) [reply docs/architecture.md]: I didn't know about this, pretty cool. Done.

File: website/docusaurus.config.js (hunk 2 above)
Comment (scottilee) [reply website/docusaurus.config.js]: Done.

File: website/package.json (hunk 3 above)
Comment (scottilee) [reply website/package.json]: Fixed.

File: website/src/css/custom.css (hunk 4 above)
Comment (scottilee) [reply website/src/css/custom.css]: Done.

**Pull request closed without merging**